### ISSUE TITLE
QT6: Crash when clicking in Locator widget (in message bar)

### ISSUE THREAD

**Issue by rouault** [qgis/QGIS#56546]
### What is the bug or the crash?

When clicking in the Locator widget  (in message bar), QGIS crashes on a QT6 build of latest master (at c32c5a31ec1e66aaebab2f3c37be5518a5074574), with Fedora Rawhide

### Steps to reproduce the issue

Stack trace:
```
#0  0x00007f5c760787cd in parsePass1 () at /usr/lib64/python3.12/site-packages/PyQt6/sip.cpython-312-x86_64-linux-gnu.so
#1  0x00007f5c7607a5d5 in parseKwdArgs () at /usr/lib64/python3.12/site-packages/PyQt6/sip.cpython-312-x86_64-linux-gnu.so
#2  0x00007f5c7607a911 in sip_api_parse_kwd_args () at /usr/lib64/python3.12/site-packages/PyQt6/sip.cpython-312-x86_64-linux-gnu.so
#3  0x00007f5c1b29759d in meth_QgsLocatorResult_setUserData (sipSelf=0x0, sipArgs=0x7f5c16da75c0, sipKwds=0x0) at /home/even/qgis/qgis/build_fedora_qt6/python/core/build/_core/sip_corepart15.cpp:16073
#4  0x00007f5d6e4de016 in cfunction_call () at /lib64/libpython3.12.so.1.0
#5  0x00007f5d6e4b4dd6 in _PyObject_MakeTpCall () at /lib64/libpython3.12.so.1.0
#6  0x00007f5d6e56145f in property_descr_set () at /lib64/libpython3.12.so.1.0
#7  0x00007f5d6e4ad72e in _PyObject_GenericSetAttrWithDict () at /lib64/libpython3.12.so.1.0
#8  0x00007f5d6e4ad530 in PyObject_SetAttr () at /lib64/libpython3.12.so.1.0
#9  0x00007f5d6e4c1835 in _PyEval_EvalFrameDefault () at /lib64/libpython3.12.so.1.0
#10 0x00007f5d6e5008a5 in method_vectorcall () at /lib64/libpython3.12.so.1.0
#11 0x00007f5c7607d89e in call_method () at /usr/lib64/python3.12/site-packages/PyQt6/sip.cpython-312-x86_64-linux-gnu.so
#12 0x00007f5c7607d999 in sip_api_call_procedure_method () at /usr/lib64/python3.12/site-packages/PyQt6/sip.cpython-312-x86_64-linux-gnu.so
#13 0x00007f5c1a593d27 in sipVH__core_446
    (sipGILState=PyGILState_UNLOCKED, sipErrorHandler=0x7f5c6c5a7d20 <sipVEH_QtCore_PyQt6(_sipSimpleWrapper*, PyGILState_STATE)>, sipPySelf=0x7f5c16e6afd0, sipMethod=0x7f5c16dcc100, a0=..., a1=..., a2=0x7e12f80) at /home/even/qgis/qgis/build_fedora_qt6/python/core/build/_core/sip_corepart0.cpp:25772
#14 0x00007f5c1b6e6919 in sipQgsLocatorFilter::fetchResults (this=0x3906080, a0=..., a1=..., a2=0x7e12f80) at /home/even/qgis/qgis/build_fedora_qt6/python/core/build/_core/sip_corepart17.cpp:124090
#15 0x00007f5d6c834120 in QgsLocator::fetchResults (this=0x4faf500, string=..., c=..., feedback=0x7e12f80) at /home/even/qgis/qgis/src/core/locator/qgslocator.cpp:213
#16 0x00007f5d6c850854 in QgsLocatorModelBridge::performSearch (this=0x4fbd3f0, text=...) at /home/even/qgis/qgis/src/core/locator/qgslocatormodelbridge.cpp:120
#17 0x00007f5d72725a24 in QgsLocatorWidget::performSearch (this=0x4fbb3b0) at /home/even/qgis/qgis/src/gui/locator/qgslocatorwidget.cpp:231
#18 0x00007f5d72725ae8 in QgsLocatorWidget::triggerSearchAndShowList (this=0x4fbb3b0) at /home/even/qgis/qgis/src/gui/locator/qgslocatorwidget.cpp:244
#19 0x00007f5d7272de6f in QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, void (QgsLocatorWidget::*)()>::call(void (QgsLocatorWidget::*)(), QgsLocatorWidget*, void**)
    (f=(void (QgsLocatorWidget::*)(QgsLocatorWidget * const)) 0x7f5d72725a8e <QgsLocatorWidget::triggerSearchAndShowList()>, o=0x4fbb3b0, arg=0x7ffd358b0a30)
    at /usr/include/qt6/QtCore/qobjectdefs_impl.h:145
#20 0x00007f5d7272d2fc in QtPrivate::FunctionPointer<void (QgsLocatorWidget::*)()>::call<QtPrivate::List<>, void>(void (QgsLocatorWidget::*)(), QgsLocatorWidget*, void**)
    (f=(void (QgsLocatorWidget::*)(QgsLocatorWidget * const)) 0x7f5d72725a8e <QgsLocatorWidget::triggerSearchAndShowList()>, o=0x4fbb3b0, arg=0x7ffd358b0a30)
    at /usr/include/qt6/QtCore/qobjectdefs_impl.h:182
#21 0x00007f5d7272c891 in QtPrivate::QCallableObject<void (QgsLocatorWidget::*)(), QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)
    (which=1, this_=0x4fd3380, r=0x4fbb3b0, a=0x7ffd358b0a30, ret=0x0) at /usr/include/qt6/QtCore/qobjectdefs_impl.h:520
#22 0x00007f5d64579684 in void doActivate<false>(QObject*, int, void**) () at /lib64/libQt6Core.so.6
#23 0x00007f5d6458878d in QTimer::timeout(QTimer::QPrivateSignal) () at /lib64/libQt6Core.so.6
#24 0x00007f5d6456949f in QObject::event(QEvent*) () at /lib64/libQt6Core.so.6
#25 0x00007f5d66ea6f68 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /lib64/libQt6Widgets.so.6
#26 0x00007f5d6cd55cad in QgsApplication::notify (this=0x7ffd358b1f60, receiver=0x4fbb420, event=0x7ffd358b0d80) at /home/even/qgis/qgis/src/core/qgsapplication.cpp:602
#27 0x00007f5d64516218 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /lib64/libQt6Core.so.6
#28 0x00007f5d646cac46 in QTimerInfoList::activateTimers() () at /lib64/libQt6Core.so.6
#29 0x00007f5d647edae9 in timerSourceDispatch(_GSource*, int (*)(void*), void*) () at /lib64/libQt6Core.so.6
#30 0x00007f5d6033626c in g_main_context_dispatch_unlocked.lto_priv () at /lib64/libglib-2.0.so.0
#31 0x00007f5d603972a8 in g_main_context_iterate_unlocked.isra () at /lib64/libglib-2.0.so.0
#32 0x00007f5d603376e3 in g_main_context_iteration () at /lib64/libglib-2.0.so.0
#33 0x00007f5d647edde3 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt6Core.so.6
#34 0x00007f5d64523053 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib64/libQt6Core.so.6
#35 0x00007f5d6451effc in QCoreApplication::exec() () at /lib64/libQt6Core.so.6
#36 0x00000000002672d3 in main (argc=1, argv=0x7ffd358b4d68) at /home/even/qgis/qgis/src/app/main.cpp:1834
```

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta charset="utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: "\2610"; }
li.checked::marker { content: "\2612"; }
</style></head><body>

QGIS version | 3.37.0-Master | QGIS code revision | unknown
-- | -- | -- | --
Compiled against Qt | 6.6.1 | Running against Qt | 6.6.2
Compiled against Python | 3.12.1 | Running against Python | 3.12.2
Compiled against GDAL/OGR | 3.8.3 | Running against GDAL/OGR | 3.8.4
PROJ version | 9.3.1
EPSG Registry database version | v10.098 (2023-11-24)
GEOS version | 3.12.1-CAPI-1.18.1
Compiled against SQLite | 3.45.0 | Running against SQLite | 3.45.1
Compiled against PDAL | 2.6.2 | Running against PDAL | 2.6.3
PostgreSQL client version | 16.1
SpatiaLite version | 5.1.0
QWT version | 6.2.0
QScintilla2 version | 2.13.4
OS version | Fedora Linux 41 (Container Image Prerelease)
  |   |   |  
This copy of QGIS writes debugging output.
  |   |   |  
Active Python plugins
processing | 2.12.99

</body></html>QGIS version
3.37.0-Master
QGIS code revision
[unknown](https://github.com/qgis/QGIS/commit/unknown)
Compiled against Qt
6.6.1
Running against Qt
6.6.2
Compiled against Python
3.12.1
Running against Python
3.12.2
Compiled against GDAL/OGR
3.8.3
Running against GDAL/OGR
3.8.4
PROJ version
9.3.1
EPSG Registry database version
v10.098 (2023-11-24)
GEOS version
3.12.1-CAPI-1.18.1
Compiled against SQLite
3.45.0
Running against SQLite
3.45.1
Compiled against PDAL
2.6.2
Running against PDAL
2.6.3
PostgreSQL client version
16.1
SpatiaLite version
5.1.0
QWT version
6.2.0
QScintilla2 version
2.13.4
OS version
Fedora Linux 41 (Container Image Prerelease)

This copy of QGIS writes debugging output.

Active Python plugins
processing
2.12.99

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by nyalldawson** [comment]
This will need a similar fix to https://github.com/qgis/QGIS/pull/56332 in QgsLocatorResult

**Comment by rouault** [comment]
> This will need a similar fix to #56332 in QgsLocatorResult

thanks for the hint! fix in https://github.com/qgis/QGIS/pull/56550
